src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -38,20 +37,63 @@ namespace Microsoft.CodeAnalysis.LanguageServer.Handler
     /// and any consumers observing the results of the task returned from <see cref="ExecuteAsync{TRequestType, TResponseType}(bool, IRequestHandler{TRequestType, TResponseType}, TRequestType, ClientCapabilities, string?, CancellationToken)"/>
     /// will see the results of the handling of the request, whenever it occurred.
     /// </para>
+    /// <para>
+    /// Exceptions in the handling of non-mutating requests are sent back to callers. Exceptions in the processing of
+    /// the queue will close the LSP connection so that the client can reconnect. Exceptions in the handling of mutating
+    /// requests will also close the LSP connection, as at that point the mutated solution is in an unknown state.
```

Comment:
should we put ourselves into a failure state, and just reject all future calls from that point until we resync?

Reply:
That is what happens via the `Contract.ThrowIfNull` in `ExecuteAsync`, but I will clarify the comment.